platforms/core-configuration/core-serialization-codecs/src/main/kotlin/org/gradle/internal/serialize/codecs/core/ConfigurableFileTreeCodec.kt

```diff
@@ -30,14 +32,17 @@ class ConfigurableFileTreeCodec(
     private val fileCollectionFactory: FileCollectionFactory
 ) : Codec<ConfigurableFileTree> {
     override suspend fun WriteContext.encode(value: ConfigurableFileTree) {
+        value as DefaultConfigurableFileTree
+        write(value.resolver)
         writeFile(value.dir)
         write(value.patterns)
     }
 
     override suspend fun ReadContext.decode(): ConfigurableFileTree {
+        val resolver = read() as PathToFileResolver
```

Comment:
💅 `readNonNull<PathToFileResolver>()`

Reply:
Ah, interesting. I didn't discover this method. A couple of codecs I looked at didn't use it, and we have plenty of codecs that use `read() as ...` pattern.